src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ImmutableDictionaryOfTKeyTValueConverter.cs

```diff
@@ -20,6 +20,9 @@ protected sealed override void Add(TKey key, in TValue value, JsonSerializerOpti
         internal sealed override bool CanHaveMetadata => false;
 
         internal override bool SupportsCreateObjectDelegate => false;
+
+        internal sealed override bool IsImmutableType => true;
```

Comment:
I searched for all converters implementing the `ConvertCollection` method and there seem to be a few more that need to be covered: `ArrayConverter`, `FSharpList/FSharpSet`, `MemoryConverter` and `ReadOnlyMemoryConverter`.

I would suggest collocating the override of the property with the override of `ConvertCollection` (potentially documenting the `ConvertCollection` method itself that this property should also be overridden).

Reply:
Ok.
Thank you for your detailed guidelines.